gcloud-java-datastore/src/main/java/com/google/gcloud/datastore/GqlQuery.java

```diff
@@ -379,9 +379,11 @@ protected void populatePb(com.google.datastore.v1beta3.RunQueryRequest.Builder r
   }
 
   @Override
-  protected GqlQuery<V> nextQuery(com.google.datastore.v1beta3.QueryResultBatch responsePb) {
-    // See issue #17
-    throw new UnsupportedOperationException("paging for this query is not implemented yet");
+  protected StructuredQuery<V> nextQuery(com.google.datastore.v1beta3.RunQueryResponse responsePb) {
+    return new StructuredQuery.Builder<>(type())
```

Comment:
I fixed the `namespace` issue, but I kept the original `mergeFrom` call instead of the `fromPb` because the `fromPb` returns a StructuredQuery of unknown type `?`, whereas `nextQuery` has a type.